includes/class-fs-api.php

```diff
@@ -226,26 +228,51 @@ private function _call( $path, $method = 'GET', $params = array(), $retry = fals
 
                 if ( null !== $result &&
                      isset( $result->error ) &&
-                     isset( $result->error->code ) &&
-                     'request_expired' === $result->error->code
+                     isset( $result->error->code )
                 ) {
-                    if ( ! $retry ) {
-                        $diff = isset( $result->error->timestamp ) ?
-                            ( time() - strtotime( $result->error->timestamp ) ) :
-                            false;
-
-                        // Try to sync clock diff.
-                        if ( false !== $this->_sync_clock_diff( $diff ) ) {
-                            // Retry call with new synced clock.
-                            return $this->_call( $path, $method, $params, true );
+                    $retry = false;
+
+                    if ( ! $in_retry ) {
+                        if ( 'request_expired' === $result->error->code ) {
+                            $diff = isset( $result->error->timestamp ) ?
+                                ( time() - strtotime( $result->error->timestamp ) ) :
+                                false;
+
+                            // Try to sync clock diff.
+                            if ( false !== $this->_sync_clock_diff( $diff ) ) {
+                                // Retry call with new synced clock.
+                                $retry = true;
+                            }
+                        } else if (
+                            Freemius_Api_WordPress::IsHttps() &&
+                            ! empty( $result->error->message ) &&
+                            ( false !== strpos( $result->error->message, 'ssl' ) ) &&
+                            ( false !== strpos( $result->error->message, 'curl error 35' ) )
+                        ) {
+                            $force_http = true;
+                            $retry      = true;
+                        }
+                    }
+
+                    if ( $retry ) {
+                        if ( $force_http ) {
+                            self::$_options->set_option( 'api_force_http', true, true );

```

Comment:
@fajardoleo I think we also need to call `Freemius_Api_WordPress::SetHttp();`. If I'm right, probably best to create a `toggle_force_http( $is_http )` private helper method.